tcpdump.c

```diff
@@ -1162,17 +1162,26 @@ _U_
 		free(host_url);
 	} else
 #endif
-	status = pcap_findalldevs(&devlist, ebuf);
-	if (status < 0)
-		error("%s", ebuf);
+        status = pcap_findalldevs(&devlist, ebuf);
+        if (status < 0) {
+  		      if (devlist != NULL) {
+	  		        pcap_freealldevs(devlist);
+		        }
+		        error("%s", ebuf);
+	          return NULL;
```

Comment:
Note that `error()` never returns, so that `return` statement will never be executed.